cmd/synthetic-monitoring-agent/main.go

```diff
@@ -176,6 +173,24 @@ func run(args []string, stdout io.Writer) error {
 		zerolog.SetGlobalLevel(zerolog.ErrorLevel)
 	}
 
+	baseCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var tracerProvider trace.TracerProvider = noop.NewTracerProvider()
+	if os.Getenv("OTEL_EXPORTER_OTLP_ENDPOINT") != "" || os.Getenv("OTEL_EXPORTER_OTLP_TRACES_ENDPOINT") != "" {
```

Comment:
> This is still a problem.

I don't see the problem. What you suggest is what I would do if this SDK was used in vacuum, but it is not. The entire ecosystem is build around the expectation that you set those environment variables and it works. It's not just the trace exporter, it's the sampling algorithm, parameters, all of it is configured automatically from environment variables. We would be going out of our way to change how one thing is configured, and by doing that we make that thing inconsistent with the rest of the ecosystem, and also with the rest of the things that can be configured in this very project. I really don't think that's a good idea.

> you to realize how problematic the functional options pattern is

I do, I don't like it either. Big fan of opts structs myself.

Reply:
I've reworked the stuff. At this point it didn't make sense to keep with the `fixup!`s, so I've rebased the entire thing. Changes roughly are:
- Objects now receive the `TracerProvider` normally, not through the context. I can hardly find words for how painful it was to add this on everywhere.
- Root spans are now initiated by the scraper and the adhoc handler. Probers no longer know what a trace is. This means we lose the ability to record probe errors in a span, as Probers hide that behind the `success` boolean. We can create a child span from the Prober just with that info, but that would be duplicating it on every prober. Your call.

Reply:
> The entire ecosystem is build around the expectation that you set those environment variables and it works.

The ecosystem is built around an ill-defined notion of "ease of use". I have had to debug issues with Otel-using libraries that obtain configuration from the environment "because it's there".

I do not want stuff happening behind our backs, and more importantly I do not want stuff happening behind our user's backs. I don't want users accidentally pushing traces. If they do want to push traces, they can explicitly ask for it.

I have added a commit on top of this PR. If you are not OK with that, remove it.